packages/react-core/src/components/Page/Page.tsx

```diff
@@ -116,19 +119,32 @@ export class Page extends React.Component<PageProps, PageState> {
   componentWillUnmount() {
     const { isManagedSidebar, onPageResize } = this.props;
     if (isManagedSidebar || onPageResize) {
-      window.removeEventListener('resize', debounce(this.handleResize, 250));
+      window.removeEventListener('resize', this.handleResize);
+      document.removeEventListener('mousedown', this.handleClickMobile);
+      document.removeEventListener('touchstart', this.handleClickMobile);
     }
   }
 
-  handleResize = () => {
-    const { onPageResize } = this.props;
-    const windowSize = window.innerWidth;
+  isMobile = () =>
     // eslint-disable-next-line radix
-    const mobileView = windowSize < Number.parseInt(globalBreakpointXl.value, 10);
+    window.innerWidth < Number.parseInt(globalBreakpointXl.value, 10);
+
+  handleResize = debounce(() => {
+    const { onPageResize } = this.props;
+    const mobileView = this.isMobile();
     if (onPageResize) {
-      onPageResize({ mobileView, windowSize });
+      onPageResize({ mobileView, windowSize: window.innerWidth });
     }
     this.setState({ mobileView });
+  }, 250);
+
+  handleClickMobile = (ev: any) => {
+    if (this.isMobile() && this.outerRef.current) {
```

Comment:
can also add an additional check here 
```suggestion
    if (this.isMobile() && !this.state.mobileIsNavOpen && this.outerRef.current) {
```